### ISSUE TITLE
Fleet Server units appears to be stuck in `(CONFIGURING) Re-configuring` state when policy is updated

### ISSUE THREAD

**Issue by ycombinator** [elastic/fleet-server#3151]
Please post all questions and issues on https://discuss.elastic.co/c/beats
before opening a Github Issue. Your questions will reach a wider audience there,
and if we confirm that there is a bug, then you can open a new issue.

For security vulnerabilities please only send reports to security@elastic.co.
See https://www.elastic.co/community/security for more information.

Please include configurations and logs if available.

For confirmed bugs, please report:
- Version: `8.12.0-SNAPSHOT` (but also reproducible on 8.11.1)
- Steps to Reproduce: 

1. Start up the stack.
   ```
   elastic-package stack up --version 8.12.0-SNAPSHOT -d
   ```
2. Check the Fleet Server's Agent status.
   ```
   docker exec -it elastic-package-stack-fleet-server-1 elastic-agent status
   ```
4. Wait till Fleet Server's Agent status reports the Fleet Server component as `HEALTHY`
5. Navigate to the Fleet UI and open the Fleet Server policy.
6. Edit the Settings and toggle "Collect agent logs".
7. Save and apply the configuration changes.
8. Check the Fleet Server's Agent status again. 

Observed: The Fleet Server's Agent status continues to report the Fleet Server component units as `(CONFIGURING) Re-configuring`

Expected: Eventually the Fleet Server's Agent status reports the Fleet Server component as `HEALTHY`

**Comment by juliaElastic** [comment]
The issue was first reported on version 8.11.1 fleet-server

**Comment by ycombinator** [comment]
@bczifra Indeed, I did 🤦.  Fixed it it now.

**Comment by michel-laterman** [comment]
@ycombinator, was this a false alarm, or does it need to be investigated?

**Comment by ycombinator** [comment]
It needs to be investigated.

**Comment by juliaElastic** [comment]
I think it's a low impact issue, as it's only causing [confusion](https://github.com/elastic/sdh-beats/issues/4159#issuecomment-1845336990). Perhaps we can set the unit state to healthy when the parent component goes to healthy.

**Comment by juliaElastic** [comment]
I tried to reproduce the issue on latest fleet-server (8.13-SNAPSHOT), but the re-configuring unit state went back to healthy after a few minutes. 
I've also looked at the code and I haven't really found where the fleet-server-default component state is being set to healthy (so we could also override the unit state). Any ideas where to fix this?

**Comment by michel-laterman** [comment]
I'm unable to recreate with the latest 8.13-SNAPSHOT
`"Image": "sha256:5b83eb5d5debdab60c104262b90b13a1a92baa8960f0706767ac842af1aed79c"`
fleet-server build hash: `920d73d3` which is from is a [commit from dec 14](https://github.com/elastic/fleet-server/commit/920d73d3876807c507d8f7a723ea48a7892a48d1)

**Comment by michel-laterman** [comment]
Yes, I've tried with toggling metrics/logs monitoring. Are you still able to recreate?

**Comment by juliaElastic** [comment]
I couldn't reproduce on latest either (the configuring state went back to healthy after a few minutes), though I haven't tried on 8.12 or 8.11 as reported in the issue.

**Comment by michel-laterman** [comment]
I have not been able to reproduce on the released v8.12.0, but have been able to reproduce on 8.11.4

**Comment by jlind23** [comment]
If this is not reproducible on any 8.12 build then I believe we should close this and reopen later if needed. What do you think?
